lib/ansible/modules/cloud/openstack/os_router.py

```diff
@@ -436,7 +436,7 @@ def main():
                     cloud.add_router_interface(router, port_id=int_p_id)
                 changed = True
             else:
-                if _needs_update(cloud, module, router, net, subnet_internal_ids, internal_portids):
+                if _needs_update(cloud, module, router, net, subnet_internal_ids, internal_portids, filters):
```

Comment:
nit: I think this should be wrapped. 121 line length is longer than it should be. While Ansible has a lot of code which doesn't adhere to pep8 we should strive to be better, so lets try and keep the line length to 80 or 120.